### PR TITLE
drivers: sensor: isl29035: Update driver to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by benjaminbjornsson**

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

File: drivers/sensor/isl29035/isl29035_trigger.c [hunk 1]
```diff
@@ -154,6 +153,8 @@ int isl29035_trigger_set(const struct device *dev,
 
 	/* enable interrupt callback */
 	setup_int(dev, true);
+	if (gpio_pin_get_dt(&config->int_gpio) > 0) {
+		handle_int(dev);
```
Comment (MaureenHelm) [inline drivers/sensor/isl29035/isl29035_trigger.c:157]: This changes the driver behavior. Is this change intentional, and if so, can you explain the reason in the commit message?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/isl29035/isl29035_trigger.c (hunk 1 above)
Comment (benjaminbjornsson) [reply drivers/sensor/isl29035/isl29035_trigger.c:157]: This was not intentional, thank you! It's fixed now.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/isl29035/isl29035.c [hunk 2]
```diff
@@ -142,8 +143,12 @@ static int isl29035_init(const struct device *dev)
 	return 0;
 }
 
-struct isl29035_driver_data isl29035_data;
+static struct isl29035_driver_data isl29035_data_inst;
 
-DEVICE_DT_INST_DEFINE(0, &isl29035_init, NULL,
-		    &isl29035_data, NULL, POST_KERNEL,
-		    CONFIG_SENSOR_INIT_PRIORITY, &isl29035_api);
+static struct isl29035_config isl29035_config_inst = {
```
[the remark this answers is not in the document]
Comment (benjaminbjornsson) [reply drivers/sensor/isl29035/isl29035.c]: Fixed!

File: drivers/sensor/isl29035/isl29035_trigger.c [hunk 3]
```diff
@@ -92,17 +93,18 @@ static void isl29035_gpio_callback(const struct device *dev,
 
 
 	ARG_UNUSED(pins);
-	handle_int(drv_data);
+	handle_int(drv_data->dev);
```
Comment (galak) [inline drivers/sensor/isl29035/isl29035_trigger.c:96]: can you just pass in `dev` and remove `drv_data`?

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
